data-prepper-plugins/opensearch/src/test/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/IndexConfigurationTests.java

```diff
@@ -71,6 +78,33 @@ public void testValidCustom() throws MalformedURLException {
         assertEquals(-1, indexConfiguration.getBulkSize());
     }
 
+    @Test
+    public void testValidCustom_from_s3() {
+        final String testTemplateFilePath = "s3://folder/file.json";
+        final String testS3AwsRegion = "us-east-2";
+        final String testS3StsRoleArn = "arn:aws:iam::123456789:user/user-role";
+
+        final S3Client s3Client = mock(S3Client.class);
+        when(s3Client.getObject(any(GetObjectRequest.class))).thenReturn(null);
+
+        final GetObjectAttributesResponse getObjectAttributesResponse = GetObjectAttributesResponse.builder().objectSize(100L).build();
+        when(s3Client.getObjectAttributes(any(GetObjectAttributesRequest.class))).thenReturn(getObjectAttributesResponse);
+
+        final String testIndexAlias = "foo";
```

Comment:
It's better to use random strings. It helps clarify that the exact value doesn't matter.